scripts/dataset-fetch.py

```diff
@@ -38,12 +38,19 @@ def wget(cfg, batch, index):
     print "Fetching batch %d" % index
     urls = [cfg.url + chunk for chunk in batch]
     pchunk = int(math.ceil(len(urls)/ncpus))
-    pargs = (urls[x:x+pchunk] for x in range(0, len(urls), pchunk))
+    pargs = [urls[x:x+pchunk] for x in range(0, len(urls), pchunk)]
     procs = [wget1(cfg, args) for args in pargs]
-    for p in procs:
+    retry = []
+    for x, p in enumerate(procs):
         rc = p.wait()
         if rc != 0:
-            raise Exception("Error fetching data: wget exit code %d" % rc)
+            retry += pargs[x]
```

Comment:
I see we are continuing to reimplement `parallel` 😄  (`--retry-failed`, in this case)